frameworks/Python/web2py/app/standard/modules/database.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+import os
+from operator import itemgetter
+from gluon.storage import Storage
+from gluon.dal import DAL, Field, Row
+
+DBHOST = os.environ.get('DBHOST', 'localhost')
+DATABASE_URI = 'mysql://benchmarkdbuser:benchmarkdbpass@%s:3306/hello_world' % DBHOST
+
+class Dal(object):
+    def __init__(self, table=None, pool_size=8):
+        self.db = DAL(DATABASE_URI, migrate_enabled=False, pool_size=pool_size)
+        if table == 'World':
+            self.db.define_table('World', Field('randomNumber', 'integer'))
+        elif table == 'Fortune':
+            self.db.define_table('Fortune', Field('message'))
+
+    def get_world(self, wid):
+        return self.db(self.db.World.id == wid).select(cacheable=True)[0].as_dict()
```

Comment:
Yeah, seems like you need to actually set the "cache" attribute in order to have the results cached.

"The select method also takes a cache argument, which defaults to None. For caching purposes, it should be set to a tuple where the first element is the cache model (cache.ram, cache.disk, etc.), and the second element is the expiration time in seconds."

http://web2py.com/books/default/chapter/29/06/the-database-abstraction-layer#Caching-selects